### ISSUE TITLE
Add support for Thermostat Setback CC

### ISSUE THREAD

**Issue by marcelveldt** [zwave-js/backlog#15]
The Thermostat Setback CC allows overriding a thermostat setpoint (the way I understand it without affecting the original setpoint setting), either temporarily (e.g. on a timer) or permanently.

![grafik](https://user-images.githubusercontent.com/17641229/236856444-4b64966f-6b32-4fa7-822d-7d58a811c35c.png)

There is no way to use/set this in HA, except using the invoke_cc_api service call.

### Diagnostics information

[zwave_js-f0926e113c178046d0d249fcdc42964a-Node 67-7180da6f98ebab1058d1cfaa753baaeb.json.txt](https://github.com/home-assistant/core/files/11422162/zwave_js-f0926e113c178046d0d249fcdc42964a-Node.67-7180da6f98ebab1058d1cfaa753baaeb.json.txt)

**Comment by marcelveldt** [comment]
candidate for expert UI ?
review zigbee + matter spec if this is common there as well, if not --> make this a setting in the export UI, otherwise handle it globally

**Comment by MartinHjelmare** [comment]
What's the status here? Did we review zigbee and matter?

**Comment by marcelveldt** [comment]
> What's the status here? Did we review zigbee and matter?

I couldn't find anything for this in zigbee and matter, seems a (somewhat weird) unique feature for Z-Wave. 
So: if this is really mandatory for certification; it should be in the expert UI

**Comment by MartinHjelmare** [comment]
Do we want the expert UI to use the existing service call?

**Comment by marcelveldt** [comment]
> Do we want the expert UI to use the existing service call?

No, if we want to deprecate some of the super advanced service calls at some point that makes it harder. I suggest the expert UI issues the commands directly

**Comment by wendevlin** [comment]
@marcelveldt 
what do I need to send then as command?
`setbackState`?

and do I send -128 or -12.8?

```json
        {
          "endpoint": 0,
          "commandClass": 71,
          "commandClassName": "Thermostat Setback",
          "property": "setbackState",
          "propertyName": "setbackState",
          "ccVersion": 1,
          "metadata": {
            "type": "number",
            "readable": true,
            "writeable": true,
            "label": "Setback state",
            "min": -12.8,
            "max": 12,
            "stateful": true,
            "secret": false
          },
          "value": 0
        },
```

**Comment by AlCalzone** [comment]
> and do I send -128 or -12.8?

Z-Wave JS deals with conversion to the correct format, so you send something between -12.8 and +12. There are 3 special states though that are not encoded in this value, so it would be better to use `invoke_cc_api` instead, with this method:
https://zwave-js.github.io/node-zwave-js/#/api/CCs/ThermostatSetback?id=set

`SetbackType` looks like this:
```ts
export enum SetbackType {
	None = 0x00,
	Temporary = 0x01,
	Permanent = 0x02,
}
```

`SetbackState` is a number between -12.8 and 12, or one of these strings:
```ts
export type SetbackSpecialState =
	| "Frost Protection"
	| "Energy Saving"
	| "Unused";
```

**Comment by wendevlin** [comment]
@AlCalzone I am not able to call the set method correctly:
![image](https://github.com/user-attachments/assets/49ddcf34-aadc-41bc-8883-803ac7c424a4)

I get the error `zwave_error: Z-Wave error 322 - setbackType is not a SetbackType (ZW0322)`

**Comment by AlCalzone** [comment]
You're passing a dictionary as the only argument, but it expects two separate non-dictionary arguments

**Comment by bueeler** [comment]
As (https://github.com/home-assistant/core/issues/92808) was closed already, I try to ask here...
While I can invoke "setOverride" through "invoke_cc_api" in HomeAssistant, it is not possible to get the current value by "invoke_cc_api". It asks for parameter:

![image](https://github.com/user-attachments/assets/a0243988-8ec3-4c09-9659-8cb93c5ceba0)

However parameter is not needed according to https://zwave-js.github.io/node-zwave-js/#/api/CCs/ClimateControlSchedule?id=getoverride. 
Is there another way to get currently set override value (setback state)? Unfortunately it is not included as attribute  for the climate instance in HA.

**Comment by AlCalzone** [comment]
Just pass an empty list (`[]`) for the parameters.

**Comment by bueeler** [comment]
@AlCalzone Thanks for the help!
Calling the service works with empty list as suggested. But it seems not possible to get the return variable.
Service call:
```
alias: GetOverride
sequence:
  - action: zwave_js.invoke_cc_api
    data:
      command_class: "70"
      method_name: getOverride
      parameters: []
      device_id:
        - 8968d62173a28aae2bc680497d39f227
    response_variable: result
description: ""
```
Error msg:
> GetOverride: Error executing script. Invalid data for call_service at pos 1: Script does not support 'response_variable' for service 'response_variable' which does not support response data.

Is there something wrong in my script, or does zwave_js.invoke_cc_api not work as expected?

**Comment by AlCalzone** [comment]
Looks like this is done, but I can't fully verify yet because the certification testing tool crashes.
